pkg/expression/constant_propagation.go

```diff
@@ -348,12 +413,21 @@ func (s *propConstSolver) propagateColumnEQ() {
 			if lOk && rOk && lCol.GetType(s.ctx.GetEvalCtx()).GetCollate() == rCol.GetType(s.ctx.GetEvalCtx()).GetCollate() && !lCol.GetType(s.ctx.GetEvalCtx()).Hybrid() && !rCol.GetType(s.ctx.GetEvalCtx()).Hybrid() {
 				lID := s.getColID(lCol)
 				rID := s.getColID(rCol)
-				s.unionSet.Union(lID, rID)
 				visited[i] = true
+				if s.unionSet.FindRoot(lID) != s.unionSet.FindRoot(rID) {
+					// Add the equality relation to unionSet
+					// if it has been added, we don't need to process it again.
+					// It will be deleted in the replaceEqCondtionWithTrueitionsWithConstants
+					s.unionSet.Union(lID, rID)
+				} else if lID != rID {
+					s.conditions[i] = &Constant{
```

Comment:
here means LID and RID is already in the union set(the equiv relationship is catched, so we can safely convert this EQ(LID, RID) to be a true)

what i concern here is that, next time, if we wanna do the constant prop one more time, we should build the union set from these condition slice again, while may we still be able to catch the equiv relation ship between LID and RID?

Reply:
Actually, it doesn't affect the logic here, since all equivalence relations are preserved, and only the redundant equivalence relations are eliminated.